File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/shortcut/ShortcutSpell.scala





















// Generated on 05/03/2013 10:35:08
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ShortcutSpell extends Shortcut {
  override val typeId = 368

  protected trait Type extends Shortcut.Type with Serializable {
    override val typeId = ShortcutSpell.typeId

    val spellId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(spellId)
    }

  }

  protected def create(slot: Int, spellId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val slot = buf.readInt()
    val spellId = buf.readShort()

    create(slot, spellId)
  }

}

object ShortcutSpell extends ShortcutSpell {
  case class TypeImpl(slot: Int, spellId: Short) extends Type
  override def create(slot: Int, spellId: Short) = TypeImpl(slot, spellId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(slot: Int, spellId: Short) = create(slot, spellId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ShortcutSpell.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ShortcutSpell extends Shortcut {
    public static final short TYPE_ID = 368;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short spellId;
    
    public ShortcutSpell() { }
    
    public ShortcutSpell(int slot, short spellId) {
        super(slot);
        this.spellId = spellId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(spellId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        spellId = buf.readShort();
        if (spellId < 0)
            throw new RuntimeException("Forbidden value on spellId = " + spellId + ", it doesn't respect the following condition : spellId < 0");
    }
    
}
